satellite/gracefulexit/chore.go

```diff
@@ -58,7 +61,47 @@ func (chore *Chore) Run(ctx context.Context) (err error) {
 			return nil
 		}
 
-		pathCollector := NewPathCollector(chore.db, exitingNodes, chore.log, chore.config.ChoreBatchSize)
+		exitingNodesLoopIncomplete := make(storj.NodeIDList, 0, nodeCount)
+		for _, node := range exitingNodes {
+			if node.ExitLoopCompletedAt == nil {
+				exitingNodesLoopIncomplete = append(exitingNodesLoopIncomplete, node.NodeID)
+				continue
+			}
+
+			progress, err := chore.db.GetProgress(ctx, node.NodeID)
+			if err != nil && !errs.Is(err, sql.ErrNoRows) {
+				chore.log.Error("error retrieving progress for node", zap.Stringer("Node ID", node.NodeID), zap.Error(err))
+				continue
+			}
+
+			lastActivityTime := *node.ExitLoopCompletedAt
+			if progress != nil {
+				lastActivityTime = progress.UpdatedAt
+			}
+
+			// check inactive timeframe
+			if lastActivityTime.Before(time.Now().UTC().Add(-chore.config.MaxInactiveTimeFrame)) {
```

Comment:
nit - `lastActivityTime.Add(chore.config.MaxInactiveTimeFrame).After(time.Now().UTC())` seems slightly easier to read